cmd/goal/application.go

```diff
@@ -176,12 +176,10 @@ func init() {
 
 	infoAppCmd.MarkFlagRequired("app-id")
 
-	methodAppCmd.MarkFlagRequired("method")      // nolint:errcheck // follow previous required flag format
-	methodAppCmd.MarkFlagRequired("app-id")      // nolint:errcheck
-	methodAppCmd.MarkFlagRequired("from")        // nolint:errcheck
-	methodAppCmd.Flags().MarkHidden("app-arg")   // nolint:errcheck
-	methodAppCmd.Flags().MarkHidden("app-input") // nolint:errcheck
-	methodAppCmd.Flags().MarkHidden("i")         // nolint:errcheck
```

Comment:
What are these and why did they go away?

Reply:
Suppose we want to support foreignAssets/foreignApp/etc., we can pass that in cmd line and I suppose they can also be passed from a file, right?
This was left from previous #3088 about if keeping argument for foreignApp etc., and I think the option should be available and kept

Reply:
The only things that went away are:
```go
methodAppCmd.Flags().MarkHidden("app-input") // nolint:errcheck
methodAppCmd.Flags().MarkHidden("i")         // nolint:errcheck
```
They are both flags which specify a JSON input file which can contain foreign references (and app args, but we already handle these properly). Since we want to allow specifying additional foreign references, I think it makes sense to keep this flag unhidden.